cmd/posix.go

```diff
@@ -1602,8 +1602,12 @@ func (s *posix) VerifyFile(volume, path string, empty bool, algo BitrotAlgorithm
 		return err
 	}
 
-	if empty && fi.Size() != 0 || !empty && fi.Size() == 0 {
-		return errFileUnexpectedSize
+	if fileSize != -1 {
+		bitrotSumsTotalSize := ceilFrac(fileSize, shardSize) * int64(h.Size())
+		expectedFileSize := bitrotSumsTotalSize + fileSize
+		if expectedFileSize != fi.Size() {
+			return errFileUnexpectedSize
+		}
```

Comment:
Note fileSize is never `-1` (since during VerifyFile the object is already created and size will be known even for compressed objects)

Note that size differs only for HighwayHash256S.

you can use this function and pass the size from the higher layer instead of calculating at the lower layer:

```
in bitrot.go:

+func bitrotShardFileSize(size int64, shardSize int64, algo BitrotAlgorithm) int64 {
+       if algo != HighwayHash256S {
+               return size
+       }
+       return ceilFrac(size, shardSize)*int64(algo.New().Size()) + size
+}
```

using this at lower layer in posix.go is also fine. Up to you.